packages/export/src/excel.js

```diff
@@ -0,0 +1,75 @@
+import _ from 'lodash/fp.js'
+import writeXlsxFile from 'write-excel-file/node'
+import { writeStreamData } from './utils.js'
+
+const convertToExcelCell = (value) => {
+  return {
+    type: String,
+    typeof: 'string',
+    wrap: true,
+    value: `${value}`,
+  }
+}
+
+let transformLabels = _.map(_.get('label'))
+
+export default ({
+  stream, // writable stream target stream
+  iterableData, // iterator for each page of an array of objects
+  // order list of which indicates the header label,
+  // display function for the field,
+  // and key of the record.
+  // [{ key: string(supports lodash dot notation), label: string, display: function(value, {key, record, transform})}...]
+  transform,
+  //TODO: support multi-line headers in excel and csv when implemented
+  headers = null, // array of strings to use as headers
+  onWrite = _.noop, // function to intercept writing a page of records
+}) => {
+  const excelData = [
+    _.map(
+      (value) => ({ value, fontWeight: 'bold' }),
+      headers || transformLabels(transform)
+    ),
+  ]
+
+  let cancel = false
+  let recordsWritten = 0
+
+  return {
+    promise: (async () => {
+      for await (let r of iterableData) {
+        if (cancel) break
+        excelData.push(
+          _.map(
+            (t) =>
+              convertToExcelCell(
+                t.display(_.get(t.key, r), {
+                  key: t.key,
+                  record: r,
+                  transform,
+                })
+              ),
+            transform
+          )
+        )
+        recordsWritten = recordsWritten + _.getOr(1, 'recordCount', r)
+        await onWrite({ recordsWritten })
+      }
+      let columns = _.map(
+        () => ({
+          width: 50,
```

Comment:
What is the default behavior for width? If the default behavior is not smart maybe we should consider setting the width to the length of the column name.

Reply:
default is not smart, 50 seemed like a reasonable size to start with and wrapping the data set to true.

Reply:
I wrote some code to keep track of the longest value and then use the column name as the shortest